pgmpy/inference/Sampling.py

```diff
@@ -132,17 +142,30 @@ def rejection_sample(self, evidence=None, size=1):
         """
         if evidence is None:
             return self.forward_sample(size)
-        sampled = DataFrame(columns=self.topological_order)
+        types = [(var_name, 'int') for var_name in self.topological_order]
+        sampled = np.zeros(0, dtype=types).view(np.recarray)
         prob = 1
-        while len(sampled) < size:
-            _size = int(((size - len(sampled)) / prob) * 1.5)
+        i = 0
+        while i < size:
+            _size = int(((size - i) / prob) * 1.5)
             _sampled = self.forward_sample(_size)
+
             for evid in evidence:
-                _sampled = _sampled[_sampled.ix[:, evid[0]] == evid[1]]
+                _sampled = _sampled[_sampled[evid[0]] == evid[1]]
+
             prob = max(len(_sampled) / _size, 0.01)
-            sampled = sampled.append(_sampled)
-        sampled.reset_index(inplace=True, drop=True)
-        return sampled[:size]
+
+            if HAS_PANDAS is True:
```

Comment:
@ankurankan Here is what I'm thinking. We should set default value to recarry arrays. Now if user give 'dataframe' as an argument and he doesn't have a installation of pandas then it will throw error that pandas is not installed. Falling back isn't a good option here. This can also happen in case of default value being set to None. Or the best option IMO will be to just return recarray object. Since recarry arrays can be transformed into DataFrame easily (in a single line) this should not be onerous task for user. This way things will be uniform and will not depend on environment installations or user end carelessness.